### PR TITLE
Add infrastructure for staging database

### PR DESCRIPTION
**Pull request by aliveni**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/BPS-1220

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: infrastructure/main.tf [hunk 1]
```diff
@@ -128,7 +141,7 @@ data "azurerm_key_vault" "s2s_key_vault" {
 resource "azurerm_key_vault_secret" "POSTGRES-USER" {
   key_vault_id = "${module.send-letter-key-vault.key_vault_id}"
   name         = "${var.component}-POSTGRES-USER"
-  value        = "${module.db.user_name}"
+  value        = "${module.staging-db.user_name}"
```
Comment (doncem) [inline infrastructure/main.tf:144]: accident?

Comment (aliveni) [reply infrastructure/main.tf]: yeah, updated now

File: infrastructure/main.tf [hunk 2]
```diff
@@ -197,3 +210,35 @@ data "azurerm_key_vault_secret" "encryption_public_key" {
   name         = "encryption-public-key"
   key_vault_id = "${module.send-letter-key-vault.key_vault_id}"
 }
+
+# region staging DB secrets
+resource "azurerm_key_vault_secret" "STAGING_POSTGRES_USER" {
+  key_vault_id = "${module.send-letter-key-vault.key_vault_id}"
+  name         = "${var.component}-STAGING-POSTGRES-USER"
```
Comment (nitinprabhu) [inline infrastructure/main.tf:217]: would be nice to use same naming convention as we have used in other repos for staging

Comment (aliveni) [reply infrastructure/main.tf:217]: do you mean using lower case?

Comment (nitinprabhu) [reply infrastructure/main.tf:217]: ` name         = "${var.component}-staging-db-user"`

Comment (aliveni) [reply infrastructure/main.tf:217]: I used the existing format

Comment (doncem) [reply infrastructure/main.tf:217]: i was thinking about making a comment but reasoned myself not to

**Pull request merged**